### PR TITLE
[v2] fix stack overflow error in helm template.

### PR DESCRIPTION
**Pull request by zwwhdls**
fix https://github.com/helm/helm/issues/6116

same as pr https://github.com/helm/helm/pull/7184

### CODE REVIEW

File: cmd/helm/fetch.go [hunk 1]
```diff
@@ -166,13 +166,13 @@ func (f *fetchCmd) run() error {
...
-		if fi, err := os.Stat(ud); err != nil {
+		if _, err := os.Stat(ud); err != nil {
 			if err := os.MkdirAll(ud, 0755); err != nil {
 				return fmt.Errorf("Failed to untar (mkdir): %s", err)
 			}
 
-		} else if !fi.IsDir() {
-			return fmt.Errorf("Failed to untar: %s is not a directory", ud)
+		} else {
+			return fmt.Errorf("ailed to untar: a file or directory with the name %s already exists", ud)
```
Comment (bacongobbler) [inline cmd/helm/fetch.go]: can you fix the typo here? Should be "failed"

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by hickeyma** [comment]
Hold on merging until #7184 merges.

File: cmd/helm/fetch.go (hunk 1 above)
Comment (zwwhdls) [reply cmd/helm/fetch.go]: Hi, thank you for your review. I have fixed it and add more tests for this issue.

**Review by bacongobbler — APPROVED** [review]
code LGTM pending another review on #7184

**Review by technosophos — APPROVED** [review]

**Pull request merged**